pkg/config/runtime.go

```diff
@@ -33,46 +34,119 @@ type (
 		Validated
 		// used to evaluate selectors
 		eval expr.PredicateEvaluator
+
+		// config is organized around the identityAttribute.
+		identityAttribute string
 	}
 )
 
 // newRuntime returns a runtime object given a validated config and a predicate eval.
-func newRuntime(v *Validated, evaluator expr.PredicateEvaluator) *runtime {
+func newRuntime(v *Validated, evaluator expr.PredicateEvaluator, identityAttribute string) *runtime {
 	return &runtime{
-		Validated: *v,
-		eval:      evaluator,
+		Validated:         *v,
+		eval:              evaluator,
+		identityAttribute: identityAttribute,
 	}
 }
 
-// ResolveFn is a function that returns a list of Combined configs for an
-// attribute bag and aspect kind set pair. It is used, at runtime, to retrieve
-// the set of configs that apply to an operation.
-type ResolveFn func(bag attribute.Bag, set KindSet) ([]*pb.Combined, error)
+// GetScopes returns configuration scopes that apply given a target service
+func GetScopes(target string) ([]string, error) {
+	return getK8sScopes(target)
+}
+
+// K8S_MAX_SPLIT how many components does a k8s svc have
+const k8sMaSplit = 2
+
+// getK8sScopes return k8s scopes
+// my-svc.my-namespace.svc.cluster.local --> [ global, my-namespace, my-svc ]
```

Comment:
nit: should be a parameter, since it is configurable in k8s (not all clusters have  "svc.cluster.local" pattern). Maybe define scopes to be just subdomains?